zerver/models.py

```diff
@@ -93,6 +101,7 @@ def completely_open(domain):
     return not realm.invite_required and not realm.restricted_to_domain
 
 def get_unique_open_realm():
+    # type: () -> Realm
```

Comment:
The same comment probably applies to a bunch of other functions in this PR; assuming you agree these should hvae Optional in the return value, can you review everything that might return None in here and fix them all @ashishk1994 ?